biztown/src/components/FormInputLayout.js

```diff
@@ -1,260 +1,428 @@
+      ...aboutSectionAnswerObj,
+      ...this.state.multipleChoiceData,
+      ...this.state.freeResponseData.checkSlip,
+      ...this.state.personalFinanceData,
+    };
+
+    console.log("completedForm", completedForm);
+
+    var submitPath = this.props.postTest
+      ? "postassessment/"
+      : "preassessment/";
+    DataManager.post(submitPath, completedForm).then(() => {
+      this.props.history.push("/completionpage");
+    });
+  };
+
+  validatePage = (pageObj) => {
+    for (let question of pageObj.keys) {
+      const element = document.getElementById(`${question}`);
+
+      if (document.getElementById(`${question}`).type === "radio") {
+        const section = this.state[`${pageObj.title}`];
+
+        if (!(element.name in section)) {
+          alert(`Please choose an option for ${question}.`);
+          return false;
         }
+      }
+
+      if (document.getElementById(`${question}`).value === "") {
+        const editedString =
+          question[0].toUpperCase() + question.substring(1);
+
+        alert(`${editedString} must be filled out.`);
+        return false;
+      }
     }
-    
-    setSchoolData = schoolData => {
-        this.setState({ schoolData });
-        // initialise to first school district
-        this.setInitialSchoolInfo(Object.keys(schoolData)[0]);
-    }
 
-    setInitialSchoolInfo = schoolDistrict => {
-        var schoolData = this.state.schoolData;
-        var aboutData = JSON.parse(JSON.stringify(this.state.aboutData));
-        var school = schoolData[schoolDistrict][0].id;
-        aboutData = { ...aboutData, schoolDistrict, school};
-        this.setState({ aboutData });
+    return true;
+  };
+
+  _next = (e) => {
+    // this.handleSubmit(e)
+
+    const aboutDataObj = {
+      title: "aboutData",
+      keys: [
+        "name",
+        "birthDate",
+        "grade",
+        "teacher",
+        "participation",
+        "school",
+        "schoolDistrict",
+      ],
+    };
+
+    const multipleChoiceDataObj = {
+      title: "multipleChoiceData",
+      keys: [
+        "q01_answer",
+        "q02_answer",
+        "q03_answer",
+        "q04_answer",
+        "q05_answer",
+        "q06_answer",
+        "q07_answer",
+        "q08_answer",
+        "q09_answer",
+        "q10_answer",
+      ],
+    };
+
+    const aboutMeObj = {
+      title: "personalFinanceData",
+      keys: ["q23_answer", "q24_answer", "q25_answer"],
+    };
+
+    const aboutMyFutureObj = {
+      title: "personalFinanceData",
+      keys: [
+        "q26_answer",
+        "q27_answer",
+        "q28_answer",
+        "q29_answer",
+        "q30_answer",
+        "q31_answer",
+        "q32_answer",
+      ],
+    };
+
+    let currentStep = this.state.currentStep;
+
+    const incrementStepAndUpdateState = () => {
+      currentStep++;
+      this.setState({ currentStep: currentStep });
+      e.preventDefault();
+    };
+
+    if (currentStep === 1 && this.validatePage(aboutDataObj)) {
```

Comment:
something we could do instead of repeating this code block is mayhaps having all these page objects in a list and a single if statement alongg the lines of:
```
if (this.validatePage(pageObjsList[currentStep - 1]) { ... }
```

Reply:
Implemented.